### DISCUSSION TITLE
Insecure which new coordinator I should buy

### DISCUSSION THREAD

**Discussion by Fridolin34** [Koenkk/zigbee2mqtt#5471] [Devices]
Greetings all,

recently I had to reflash my CC2531, because it was no longer recognized by the zigbee addon. This happened out of a sudden, without any changes done to the system. I went into a storage room and the lights didn't turn on automatically. Searching the net, flashing a new firmware using the CC debugger (this time 'source routing only'), repairing a couple of the devices, ... brought the system back to life, but it did cost me three hours and at least 10 hairs turn from brown to gray.

In one of the threads here, a similar issue was closed saying: "_if it happens again, buy a new and more powerful coordinator_".

And this is the point where I guess I currently am. I've checked the supported devices page, but I cannot figure out, what '_the best_' adapter is. I mean, maybe there is not 'the best car' out there, because everyone likes something different, but concerning the coordinator, I don't care that much, whether this a big or small device, it just needs to be powerful concerning the amount of entities to be referenced as well as the range to reach them. On the other side, I see that this wiki page with supported adapters is constantly updated (btw: thank you :)) and it would make sense that tomorrow the 'best device' is still listed and supported.
So, from your point of view, is there a device recommended? I mean, currently I only have 18 devices, mostly active elements/routers and a few sensors, but somehow the number is constantly growing.

The _'getting started'_ page tells me to e.g. buy the _CC2652RB_ stick. I don't know, whether is has been there, when I first visited this page and bought the CC2531, but fine. It is described as being 'very powerful'. For a friend, I bought and installed a _CC1352P-2_ with antenna/housing and it worked flawlessly. But I didn't bring this one home to figure out, whether it would be an eye-opener compared to my cc2531.

So, could someone please tell me, what the differences between all those 'very powerful' devices are? Which device would be the most reasonable buy today?

Please note, this is just meant as some kind of exchange  - not a philosophical point to argue. 

Thanks a lot,
Fridolin

**Comment by Koenkk** [comment]
The "most" recommended adapter is either the ZZH or Slaesh CC2652RB stick (https://www.zigbee2mqtt.io/getting_started/what_do_i_need.html#supported-zigbee-adapter) (they are very similar to each other and equal in terms of performance/range).

**Reply by Morphy99** [reply]
For large 200+ networks, is this still the case? I [read](https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator) the CC2538 + CC2592 can support up to 400 routes?

**Comment by Fridolin34** [comment]
Thank you for your reply.

I guess, I might try one of those. The difference seems to be a crystal, I don't know what it is good for. So it is more or less a matter of availability and costs in the end.

Regards,
Fridolin

**Reply by Koenkk** [reply]
> The difference seems to be a crystal, I don't know what it is good for.

0 difference for the user (the difference is at the manufacturer side).

> So it is more or less a matter of availability and costs in the end.

Exactly!

**Reply by DrTron** [reply]
I also replaced my CC2531 with a Slaesh CC2652RB stick.
Works good, probably will support way more devices than I'll ever have.
Just the range is not as good as I expected (may be the location), which caused rather low link qualities. A CC2530 router helped a lot here.

**Reply by Fridolin34** [reply]
Hmm...not good to hear that. Pretty much nearby my CC2531 there are two light bulbs and only a couple of meters away, new active routers are in reach. But the link quality for all devices on this floor is only averaging round about 40. I thought to increase that by buying this new coordinator, just to make the network more stable.
I have seen, while working with the CC1352P-2, that a link quality of ~200 is possible. I don't know, where the limit is, but having an average of 100 would already make me happy.

**Reply by DrTron** [reply]
On a side note, I do have some mains-powered devices such as plugs that work as routers, but those do not improve link quality one bit.
The CC2530, however, do massively.

I currently have 15 devices in total, two of them are CC2530+CC2591 routers (needed a second one to reach the Christmas Lights plug out back), and my link quality averages around 80-90 for all devices. Network is rock stable.

**Comment by domosgabor** [comment]
Hello, I would appreciate if somebody can give me some advice which direction should I choose. 
I live in a two-story house, floor units made out of ferroconcrete, so it's not the best place for any kind of wireless communication. 

Currently I have 70+ devices in my network (40 IKEA and Philips lights, several Xiaomi sensors, hue dim switches, etc.). The coordinator is a GB2530 flashed with source routing firmware, on top of this I have four CC2530+CC2591 routers (2/2 downstairs/upstairs) . I'm planning to add more battery powered sensors and switches to the network, so it can easily have a network with 120-140 devices. 
On daily basis I'm experiencing perfomance issues and some latency but I'm pretty much satisfied with the system. What I don't know how to proceed further:
a). add 5-6 additional CC2530+CC2591 router to the network to improve the stability
b.) switch to a ZZH coordinator
c.) something else

Anyhow, if I take my current setup, I can say that I have 1 coordinator, 4 routers and 40 light that acts as router. Why do I have still performance issues? Can a ZZH coordinator improve things, or should I investigate further in my network? 

Is pairing an important issue in case of a zigbee network? I mean, does it matter, where I pair a device (near to a router, next to a bulb) or not? What happens if I pair a device next to the downstairs router and I move the device to the upstairs? 

How should I build such a huge network? Start with the routers? After that add the battery powered devices? What about the bulbs? Does it make sense to have a defined order or not?

**Reply by DrTron** [reply]
First, I'd gather data on how the link quality is for your devices.
Write all of zigbee2mqtt to a log file and grep for the link quality value and the device.
Then you'll have a good idea on which devices may be having a bad connection.
In general, link quality can have values from 0-254, and every device should be solidly above 40-50.

**Reply by domosgabor** [reply]
Thanks, great idea. I'm wondering whether the GB2530 is not able to manage 70+ devices or something is not ok with the routers. I have 20+ orhpan devices according to the zigbee map, although all of them are working mostly properly.
What is not clear, why a battery powered device prefer to join to a bulb instead of a router. Downstairs router seems to be managing a lof of devices but office and livingroom routers have only 1-2 connections. 

![zigbee20210106](https://user-images.githubusercontent.com/74321012/103799267-2943a380-504b-11eb-9173-fcc8e5e2f083.png)

**Reply by DrTron** [reply]
Those maps don't tell you much about link quality. Log them over a few days and see where you are.
Look especially at the timestamps, even a battery-powered device should send link quality updates at least a few times over 24hours. If a device does not update during a few days, you can consider it offline.

**Reply by Koenkk** [reply]
> Currently I have 70+ devices in my network (40 IKEA and Philips lights, several Xiaomi sensors, hue dim switches, etc.). The coordinator is a GB2530 flashed with source routing firmware

First upgrade your coordinator, 70 is a big amount for the CC2530. It's recommended till +- max 20 devices (upgrade to e.g. zzh: https://www.zigbee2mqtt.io/getting_started/what_do_i_need.html#supported-zigbee-adapter)

**Reply by domosgabor** [reply]
Thanks a lot! 
If I upgrade to zzh, do I still need routers in the network (CC2530)? If yes, how many what is the optimal number of routers in case of 140 devices, or zzh will easily manage all of them and I don't have to deal with routers anymore?

**Reply by DrTron** [reply]
Yes, you do. Managing the devices is one thing, but range is the other. I don't know about the zzh, I have a CC2652RB, which I think is comparable, and even with a 4"-antenna the range is not great and link qualities are rather low. My two CC2530+CC2591 routers (also with antennae) help a lot, though.

**Reply by Koenkk** [reply]
> Thanks a lot!
> If I upgrade to zzh, do I still need routers in the network (CC2530)? If yes, how many what is the optimal number of routers in case of 140 devices, or zzh will easily manage all of them and I don't have to deal with routers anymore?

I still recommend using routers, but the ZZH can handle much more directly. I don't have exact numbers, try experimenting with it.

**Comment by domosgabor** [comment]
Ok, it's clear now:
- switch to a zzh adapter
- keep some CC2530 routers in the network
- instead of checking the zigbee map, log the link quality and analyze it

What about pairing? Is it important where I add a new device to the network (where it is used)? And what about the order of devices?

**Reply by DrTron** [reply]
As far as I know, no difference. I've even re-paired devices at the far end of the house.
I use a devices.yaml file to keep things neat and group them there by device class, but that may be overdoing it.
Just remember that when you switch coordinators to a zzh/CC2652RB you need to change the PAN-ID and re-pair all devices. No way around that, sadly.

**Reply by Fridolin34** [reply]
Just it seems to be a challenge to get a CC2652. A week ago, I've put myself in the cue of two tindie shops to get notified, once new devices are available. But until now, nothing, so I still have to wait ;-)

**Reply by DrTron** [reply]
Contact slaesh directly by e-mail. He was amazing to talk to and even sent one to me in the US as a letter to save postage.

**Reply by Fridolin34** [reply]
I've been to his tindie shop (https://www.tindie.com/products/slaesh/cc2652-zigbee-coordinator-or-openthread-router/), but it still says 'out of stock'. But who knows, you might be right and asking via mail won't hurt.

**Reply by simonattilahun** [reply]
Resellers also sell slaesh CC2652 Zigbee stick from stock now. See here for example: https://ewelinkstore.com/product/cc2652rb-zigbee-usb-development-stick/

**Comment by mercenaruss** [comment]
For example between CC2562R/RB and CC2652P is a big difference. The power: 5dbm against 20 dbm. CC2652P wins
